contrib/epee/include/storages/levin_abstract_invoke2.h

```diff
@@ -34,10 +34,26 @@
 #undef MONERO_DEFAULT_LOG_CATEGORY
 #define MONERO_DEFAULT_LOG_CATEGORY "net"
 
+namespace
+{
+  template<typename context_t>
+  void on_levin_traffic(const context_t &context, bool initiator, bool sent, bool error, size_t bytes, const char *category)
+  {
+    MCINFO("net.p2p.traffic", context << bytes << " bytes " << (sent ? "sent" : "received") << (error ? "/corrupt" : "")
+        << " for category " << category << " initiated by " << (initiator ? "us" : "peer"));
+  }
+  template<typename context_t>
+  void on_levin_traffic(const context_t &context, bool initiator, bool sent, bool error, size_t bytes, int command)
+  {
+    return on_levin_traffic(context, initiator, sent, error, bytes, ("command-" + std::to_string(command)).c_str());
```

Comment:
I think its preferable to not allocate a `std::string` here for this - keep cycles to a minimum with optional debugging stuff.

The levin protocol uses a `std::uint32_t` value command. So a negative command value is illegal. And noise uses "0", which is currently a unique value (I chose it for that reason). So negative values for invalid commands and zero for noise could be used to drop the `const char*` overload entirely.